packages/@ember/-internals/utils/lib/make-array.ts

```diff
@@ -29,8 +29,7 @@ const { isArray } = Array;
  @return {Array}
  @private
  */
-function makeArray<T>(obj: T[]): T[];
-function makeArray<T>(obj: T): T[];
+function makeArray<T, TT>(obj: T): T extends TT[] ? T : T extends null | undefined ? [] : [T];
```

Comment:
Two notes:

- We don't actually need the `TT` param here at all, we can just 
- the last bit should be an array of `T` rather than a one-tuple of T, I think? Just by intent, I would be surprised to get back a tuple.

```suggestion
function makeArray<T>(obj: T): T extends Array<unknown> ? T : T extends null | undefined ? [] : T[];
```

Reply:
We did need it, but the correct NativeArray types may make it unnecessary.

Reply:
Ah, I see what the issue was. Glad the fix for that type resolved it. Thank you!